overlord/snapstate/refreshhints.go

```diff
@@ -99,7 +99,7 @@ func (r *refreshHints) refresh() error {
 		return fmt.Errorf("internal error: cannot get refresh-candidates: %v", err)
 	}
 
-	setNewRefreshCandidates(r.state, hints)
+	updateRefreshCandidates(r.state, hints)
```

Comment:
we asked about all snaps in the system, so anything that is not in hints anymore needs can/be dropped